core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/CheckBoxEditorDirective.js

```diff
@@ -0,0 +1,291 @@
+            var newChoiceNumber = choiceNames.length + 1;
+            while (choiceNames.indexOf(
+                '<p>Good Option ' + newChoiceNumber + '</p>') !== -1) {
+              newChoiceNumber++;
+            }
+
+            newCustomizationArgs.choices.value.push(
+              '<p>Good Option ' + newChoiceNumber + '</p>');
+            $scope.saveCustomizationArgs({
+              newValue: newCustomizationArgs
+            });
+            // The field needs to be initialized before the broadcast is sent.
+            $timeout(function() {
+              openChoiceEditor(newChoiceIndex);
+            });
+          };
+
+          $scope.isChoiceValid = function(index, newChoiceValue) {
+            var choiceNames = angular.copy(
+              $scope.getCustomizationArgs().choices.value);
+            choiceNames.splice(index, 1);
+
+            return (
+              Boolean(newChoiceValue) &&
+              choiceNames.indexOf(newChoiceValue) === -1);
+          };
+
+          $scope.saveChoice = function(index, newChoiceValue) {
+            if (!newChoiceValue) {
+              alertsService.addWarning('Cannot save an empty choice.');
+              return StatusObjectFactory.createFailure(
+                'Cannot save an empty choice'
+              );
+            }
+
+            var newCustomizationArgs = $scope.getCustomizationArgs();
+            var choiceNames = newCustomizationArgs.choices.value;
+
+            if (newChoiceValue === choiceNames[index]) {
+              // No change has been made.
+              return StatusObjectFactory.createFailure(
+                'No change has been made'
+              );
+            }
+
+            if (choiceNames.indexOf('newChoiceValue') !== -1) {
+              alertsService.addWarning(
+                'Cannot save: this duplicates an existing choice.');
+              return StatusObjectFactory.createFailure(
+                'This duplicates an existing choice'
+              );
+            }
+
+            newCustomizationArgs.choices.value[index] = newChoiceValue;
+            $scope.saveCustomizationArgs({
+              newValue: newCustomizationArgs
+            });
+          };
+
+          $scope.deleteChoice = function(value) {
+            var newCustomizationArgs = $scope.getCustomizationArgs();
+            if(newCustomizationArgs["choices"]["value"].length === 1){
+              throw Error(
+                'Cannot delete choice when there is only 1 choice remaining.');
+            }
+            else{
+              if(newCustomizationArgs["choices"]["value"].includes(value) ){
+                  var position = newCustomizationArgs["choices"]["value"].indexOf(value);
+                  newCustomizationArgs["choices"]["value"].splice(position, 1);
+                  newCustomizationArgs.maxAllowableSelectionCount.value--;
+                  newCustomizationArgs.minAllowableSelectionCount.value--;
```

Comment:
Why is this being modified? I thought the min would always stay at 1. Do these values get updated when a choice is added?